src/main/java/budgetbuddy/model/rule/expression/ActionExpression.java

```diff
@@ -48,6 +48,11 @@ public Value getValue() {
         return value;
     }
 
+    @Override
+    public String getType() {
+        return RuleAction.TYPE_EXPRESSION;
+    }
+
```

Comment:
Why not have `getType()` in `RuleAction` return `TYPE_EXPRESSION` ? Then you don't need to override for each child class. Unless a future child class might return something different. Same for `RulePredicate` .

Reply:
Yup, this is in consideration of possible scripts that are also ruleactions/rulepredicates

Reply:
Huh, sorry I'm not understanding. If a script extends `RuleAction`, then wouldn't it inherit `getType()` and its implementation?

Reply:
So getType() is in both RulePredicate and RuleAction as abstract methods. so when a script/expression extends it, it'll have to implement getType, but perhaps with a different type. Hence only ActionExpression has TYPE_EXPRESSION.

Reply:
Oh hm ok, you're probably more aware of what's extending what so I'll leave it to you.